app/main.py

```diff
@@ -1,3 +1,51 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, dist: Distance | int | float) -> Distance:
+        if isinstance(dist, Distance):
+            return Distance(self.km + dist.km)
+        return Distance(self.km + dist)
+
+    def __iadd__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, dist: int | float) -> Distance:
+        return Distance(self.km * dist)
+
+    def __truediv__(self, dist: int | float) -> Distance:
+        return Distance(round(self.km / dist, 2))
+
+    def __lt__(self, other: Distance | int | float) -> bool:
+        if not isinstance(other, Distance):
+            other = Distance(other)
+        return self.km < other.km
+
+    def __gt__(self, other: Distance | int | float) -> bool:
+        if not isinstance(other, Distance):
+            other = Distance(other)
+        return self.km > other.km
+
+    def __eq__(self, other: Distance | int | float) -> bool:
+        if not isinstance(other, Distance):
+            other = Distance(other)
+        return self.km == other.km
+
+    def __le__(self, other: Distance | int | float) -> bool:
+        return self < other or self == other
```

Comment:
It is better to check by using this operator -> `<=`
Also why you do not check possible variants if `other` will be `Distance` type ? ( And in __ge__ method also )